modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/function/ZoomLevelFunction.java

```diff
@@ -60,7 +60,9 @@
                     parameter("scaleDenominator", Number.class),
                     parameter("srid", String.class));
 
-    public static final double EPSG_3857_O_SCALE = 559_082_263.9508929;
+    public static final int ROOT_TILE_PIXELS =
+            Integer.valueOf(System.getProperty("MBSTYLE_ROOT_TILE_PIXELS", "512"));
+    public static final double EPSG_3857_O_SCALE = 559_082_263.9508929 / ROOT_TILE_PIXELS * 256;
```

Comment:
Took me a minute to figure out this one - maybe use brackets to be a little more clear - `(559_082_263.9508929 / ROOT_TILE_PIXELS) * 256;`

Reply:
Done